tests/cbmc/proofs/s2n_socket_was_corked/s2n_socket_was_corked_harness.c

```diff
@@ -0,0 +1,26 @@
+/* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved. */
+/* SPDX-License-Identifier: Apache-2.0 */
+
+/**
+ * @file s2n_socket_was_corked_harness.c
+ * @brief Implements the proof harness for s2n_socket_was_corked function.
+ */
+
+#include <utils/s2n_socket.h>
+
+void s2n_socket_was_corked_harness()
+{
+  struct s2n_socket_write_io_context *cbmc_allocate_s2n_socket_write_io_context();
+  /* Non-deterministic inputs. */
+  struct s2n_connection *s2n_connection = malloc(sizeof(*s2n_connection));
+  if (s2n_connection != NULL) {
+    s2n_connection->send_io_context = cbmc_allocate_s2n_socket_write_io_context();
+  }
+  
+  /* Operation under verification. */
+  int result = s2n_socket_was_corked(s2n_connection);
+
+  /* Post-condition. */
+  assert(S2N_IMPLIES(s2n_connection != NULL && (!s2n_connection->managed_io || !s2n_connection->send), result == 0)); /* false */
+  assert(S2N_IMPLIES(s2n_connection == NULL, result != 0));
```

Comment:
Use `S2N_SUCCESS` in place of `0` in this line and above.

Reply:
Even though 0 signifies false here?

Reply:
My bad! `0` makes sense here. Resolving this.